core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -138,7 +139,7 @@ private[spark] class TaskSetManager(
   private[scheduler] var pendingTasksWithNoPrefs = new ArrayBuffer[Int]
 
   // Set containing all pending tasks (also used as a stack, as above).
-  private val allPendingTasks = new ArrayBuffer[Int]
+  private var allPendingTasks = new ArrayBuffer[Int]
```

Comment:
I make this `var`, because I don't have a better approach to sort ArrayBuffer in place. Advice?